### PR TITLE
rgw/pubsub: remove deprecated radogw-admin pubsub creation commands

### PR DESCRIPTION
**Pull request by yuvalif**
make topic and subscription read commands an official feature

Fixes: https://tracker.ceph.com/issues/43536

Signed-off-by: Yuval Lifshitz <ylifshit@redhat.com>

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
new CLI test is failing in teuthology:
http://qa-proxy.ceph.com/teuthology/yuvalif-2020-06-01_08:31:27-rgw-fix_issue_43536-distro-basic-smithi/5109892/teuthology.log

**Comment by cbodley** [comment]
> INFO:teuthology.orchestra.run.smithi145.stderr:ERROR: could not get topics: (2) No such file or directory

which object isn't being found? maybe the user has to create topics before that object exists?

**Comment by yuvalif** [comment]
> > INFO:teuthology.orchestra.run.smithi145.stderr:ERROR: could not get topics: (2) No such file or directory
> 
> which object isn't being found? maybe the user has to create topics before that object exists?

topic i created one line before that: https://github.com/ceph/ceph/pull/35331/files#diff-f6e18cedf532c3572a54d443f57f5106R2236

**Comment by yuvalif** [comment]
> > > INFO:teuthology.orchestra.run.smithi145.stderr:ERROR: could not get topics: (2) No such file or directory
> > 
> > 
> > which object isn't being found? maybe the user has to create topics before that object exists?
> 
> topic is created one line before that: https://github.com/ceph/ceph/pull/35331/files#diff-f6e18cedf532c3572a54d443f57f5106R2236

topic is there, added an assertion to the test to fetch the topics via the REST API, and it is working fine, but still failing when fetching them via radosgw-admin.
both the REST API and the radosgw-admin fetch are passing when running locally (even with 3 zone cluster).

**Comment by yuvalif** [comment]
ran it again with radosgw-admin debug messages: http://qa-proxy.ceph.com/teuthology/yuvalif-2020-06-03_18:30:30-rgw:multisite-master-distro-basic-smithi/5116018/teuthology.log
extracted radosgw-admin relevant log:
https://paste.sh/qZ7a9FcL#K4kf3Y9pOaEgCoFNTok73xM9

File: src/test/rgw/rgw_multi/tests_ps.py [hunk 1]
```diff
@@ -2221,6 +2222,99 @@ def test_ps_subscription():
...
+    master_zone, ps_zone = init_env()
+    bucket_name = gen_bucket_name()
+    topic_name = bucket_name+TOPIC_SUFFIX
+    realm = get_realm()
+    zonegroup = realm.master_zonegroup()
+
+    # create topic
+    topic_conf = PSTopic(ps_zone.conn, topic_name)
+    topic_conf.set_config()
+    result, status = ps_zone.zone.cluster.admin(['topic', 'list', '--uid', get_user()])
```
Comment (cbodley) [inline src/test/rgw/rgw_multi/tests_ps.py:2237]: this is missing the --rgw-zone argument. in teuthology, this pubsub zone (test-zone4) is running on the same cluster as test-zone2, and test-zone2 is set as the --default

the function `Zone.zone_arg()` returns `['--rgw-zone', self.name]`, so you can add that to your commands:
```python
cluster.admin(['topic', 'list', '--uid', get_user()] + ps_zone.zone.zone_arg())
```

**Comment by yuvalif** [comment]
radosgw-admin tests passing after latest fix:
http://pulpito.front.sepia.ceph.com/yuvalif-2020-06-08_19:57:02-rgw:multisite-master-distro-basic-smithi/
(failures are unrelated, and due to not rebasing to latest osd code)

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
jenkins render docs

**Comment by yuvalif** [comment]
jenkins test docs

**Comment by yuvalif** [comment]
jenkins test dashboard backend

**Comment by yuvalif** [comment]
latest teuthology:
http://pulpito.front.sepia.ceph.com/yuvalif-2020-06-23_14:40:15-rgw-wip-yuval-test-35331-35155-distro-basic-smithi/
* ` cluster [WRN] Health check failed: Reduced data availability: 1 pg inactive, 1 pg peering (PG_AVAILABILITY)" in cluster log` is still seen
* new issue see: `Scrubbing terminated -- not all pgs were active and clean.`
* multisite failures:
```
2020-06-24T06:48:31.540 INFO:tasks.rgw_multisite_tests:======================================================================
2020-06-24T06:48:31.540 INFO:tasks.rgw_multisite_tests:FAIL: tasks.rgw_multi.tests.test_multipart_object_sync
2020-06-24T06:48:31.541 INFO:tasks.rgw_multisite_tests:----------------------------------------------------------------------
2020-06-24T06:48:31.541 INFO:tasks.rgw_multisite_tests:Traceback (most recent call last):
2020-06-24T06:48:31.541 INFO:tasks.rgw_multisite_tests:  File "/home/teuthworker/src/git.ceph.com_git_teuthology_master/virtualenv/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
2020-06-24T06:48:31.542 INFO:tasks.rgw_multisite_tests:    self.test(*self.arg)
2020-06-24T06:48:31.542 INFO:tasks.rgw_multisite_tests:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-yuval-test-35331-35155/qa/tasks/rgw_multi/tests.py", line 1200, in test_multipart_object_sync
2020-06-24T06:48:31.542 INFO:tasks.rgw_multisite_tests:    zonegroup_bucket_checkpoint(zonegroup_conns, bucket.name)
2020-06-24T06:48:31.543 INFO:tasks.rgw_multisite_tests:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-yuval-test-35331-35155/qa/tasks/rgw_multi/tests.py", line 396, in zonegroup_bucket_checkpoint
2020-06-24T06:48:31.543 INFO:tasks.rgw_multisite_tests:    zone_bucket_checkpoint(target_conn.zone, source_conn.zone, bucket_name)
2020-06-24T06:48:31.543 INFO:tasks.rgw_multisite_tests:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-yuval-test-35331-35155/qa/tasks/rgw_multi/tests.py", line 388, in zone_bucket_checkpoint
2020-06-24T06:48:31.544 INFO:tasks.rgw_multisite_tests:    (target_zone.name, source_zone.name, bucket_name)
2020-06-24T06:48:31.544 INFO:tasks.rgw_multisite_tests:AssertionError: failed bucket checkpoint for target_zone=a2 source_zone=a1 bucket=znwidg-50
2020-06-24T06:48:31.544 INFO:tasks.rgw_multisite_tests:
```
* orphan list test is failing: http://qa-proxy.ceph.com/teuthology/yuvalif-2020-06-23_14:40:15-rgw-wip-yuval-test-35331-35155-distro-basic-smithi/5173419/teuthology.log
* seems like a test issue:
```
2020-06-24T05:13:22.135 INFO:teuthology.orchestra.run.smithi012.stderr:'CommandManager' object has no attribute 'add_command_group'
2020-06-24T05:13:22.136 INFO:teuthology.orchestra.run.smithi012.stderr:Traceback (most recent call last):
2020-06-24T05:13:22.136 INFO:teuthology.orchestra.run.smithi012.stderr:  File "/home/ubuntu/cephtest/keystone/.tox/venv/lib/python3.6/site-packages/cliff/app.py", line 264, in run
2020-06-24T05:13:22.136 INFO:teuthology.orchestra.run.smithi012.stderr:    self.initialize_app(remainder)
2020-06-24T05:13:22.136 INFO:teuthology.orchestra.run.smithi012.stderr:  File "/home/ubuntu/cephtest/keystone/.tox/venv/lib/python3.6/site-packages/openstackclient/shell.py", line 133, in initialize_app
2020-06-24T05:13:22.136 INFO:teuthology.orchestra.run.smithi012.stderr:    super(OpenStackShell, self).initialize_app(argv)
2020-06-24T05:13:22.137 INFO:teuthology.orchestra.run.smithi012.stderr:  File "/home/ubuntu/cephtest/keystone/.tox/venv/lib/python3.6/site-packages/osc_lib/shell.py", line 442, in initialize_app
2020-06-24T05:13:22.137 INFO:teuthology.orchestra.run.smithi012.stderr:    self._load_plugins()
2020-06-24T05:13:22.137 INFO:teuthology.orchestra.run.smithi012.stderr:  File "/home/ubuntu/cephtest/keystone/.tox/venv/lib/python3.6/site-packages/openstackclient/shell.py", line 104, in _load_plugins
2020-06-24T05:13:22.137 INFO:teuthology.orchestra.run.smithi012.stderr:    self.command_manager.add_command_group(cmd_group)
2020-06-24T05:13:22.137 INFO:teuthology.orchestra.run.smithi012.stderr:AttributeError: 'CommandManager' object has no attribute 'add_command_group'
```

**Pull request merged**